doc/design/zarr-manifests.md

```diff
@@ -0,0 +1,199 @@
+
+        For example, if `fields` had a value of `["versionId", "lastModified",
+        "size", "ETag"]`, then a possible entry array could be:
+
+        ```json
+        [
+            "VI067uTlzPTTyL750Ibkx3hAUm67A_sI",
+            "2022-03-16T02:39:36+00:00",
+            27935,
+            "fc3d1270cd950f1e5430226db4c38c0e"
+        ]
+        ```
+
+        Here, the first element of the array is the entry's `versionId`, the
+        second element is the entry's `lastModified` timestamp, the third
+        element is the entry's size, and the fourth entry is the entry's ETag.
+
+    - Each directory in the Zarr is represented as an object in which each key
+      is the name of an entry or subdirectory inside the directory and the
+      corresponding value is either an entry array or a directory object.
+
+    - The `entries` object itself represents the top level directory of the
+      Zarr.
+
+    For example, a Zarr with the following structure:
+
+    ```text
+    .
+    ├── .zgroup
+    ├── arr_0/
+    │   ├── .zarray
+    │   └── 0
+    └── arr_1/
+        ├── .zarray
+        └── 0
+    ```
+
+    would have an `entries` field as follows (with elements of the entry arrays
+    omitted):
+
+    ```json
+    {
+        ".zgroup": [ ... ],
+        "arr_0": {
+            ".zarray": [ ... ],
+            "0": [ ... ]
+        },
+        "arr_1": {
+            ".zarray": [ ... ],
+            "0": [ ... ]
+        }
+    }
+    ```
+
+> [!NOTE]
+> The manifest files created by @yarikoptic contain the following fields which
+> are not present in the format described above:
+>
+> - A top-level `schemaVersion` key with a constant value of `2`
+>
+> - A `zarrChecksumMismatch` field inside the `statistics` object, used to
+>   store the checksum that the API reports for a Zarr when it disagrees with
+>   the checksum calculated by the manifest-generation code
+
+
+Archive API Changes
+-------------------
+
+***WIP***
+
+* Zarr version IDs equal the Zarr checksum
+
+* Asset properties gain `zarr_version: str | null` field (absent or null if Zarr is not yet ingested or asset is not a Zarr)
+    - Not settable by client
+    - Mint new asset when version changes?
+
+* Add `zarr_version` field to …/assets/path/ results
+
+* Zarr `contentUrl`s:
+    - Make API download URLs for Zarrs redirect to dandidav
+    - Replace S3 URLs with webdav.{archive_domain}/zarr/ URLs?
+        - Document needed changes to dandidav?
+            - The bucket for the Archive instance will now be given on the command line (only required if a custom/non-default API URL is given)
+            - The bucket's region will have to be looked up & stored before starting the webserver
+            - Zarrs under `/dandisets/` will no longer determine their S3 location via `contentUrl`; instead, they will combine the Archive's bucket & region with the Zarr ID in the asset properties (templated into "zarr/{zarr_id}/")
```

Comment:
note to ourselves: requests to modify a Zarr from previous non draft version might be "hard to impossible". we should keep that in mind while developing life cycle here.

Reply:
@yarikoptic 

> requests to modify a Zarr from a previous non draft version

What are you talking about?  A "non draft version" is a published version, and published versions and their contents can't be modified.

> we need association of zarr to a dandiset

Each Zarr is already associated with a Dandiset.  You can see a Zarr's Dandiset by requesting `/zarr/{zarr_id}/`.